### PR TITLE
[Web] Fix webview not destroyed

### LINKED ISSUE #1220: [Web] WebView not destroyed on back
## Describe the bug
The WebView is not destroyed on composable dispose. For example, if a sound is played in the webpage, the sound continue to play after going back.

## To Reproduce
1. Go to BasicWebViewSample
2. Load this url https://webbrowsertools.com/audio-test/
3. Choose an audio sample (like mono 44100Hz)
4. Press back

## Expected behavior
Everything should stop and be released

## Environment:
 - Android OS version: 12
 - Device: Pixel 3XL
 - Accompanist version: 0.24.14-SNAPSHOT

### PR DESCRIPTION
**Pull request by rlatapy-luna**
Fix #1220 

Give access to the webview during composable dispose run. By default, destroy the webview.

Use https://webbrowsertools.com/audio-test/ for testing (see #1220)

### CODE REVIEW

File: web/src/main/java/com/google/accompanist/web/WebView.kt [hunk 1]
```diff
@@ -68,6 +70,7 @@ fun WebView(
     captureBackPresses: Boolean = true,
     navigator: WebViewNavigator = rememberWebViewNavigator(),
     onCreated: (WebView) -> Unit = {},
+    onDispose: (WebView) -> Unit = { it.destroy() },
```
Comment (bentrengrove) [inline web/src/main/java/com/google/accompanist/web/WebView.kt:73]: I don't mind exposing the callback but I think `it.destroy()` should be moved to onDispose and always called. Is there a valid usecase for not calling it?

**Review by bentrengrove — CHANGES_REQUESTED** [review]
Thank you for the bug report and the PR to fix it!

Could you please also add a basic test to ensure onDispose is called?

**Comment by bentrengrove** [comment]
You'll also have to run

`./gradlew metalavaGenerateSignature` to update the API file

File: web/src/main/java/com/google/accompanist/web/WebView.kt (hunk 1 above)
Comment (rlatapy-luna) [reply web/src/main/java/com/google/accompanist/web/WebView.kt:73]: I was thinking about configuration changes. But it probably requires more work to handle it properly (cf https://github.com/google/accompanist/issues/1178).

**Comment by bentrengrove** [comment]
That is a very good point about making that linked issue worse. This might actually be too big a breaking change to introduce in the rc stage. For now, let's remove the `it.destroy` from the callback and just expose the callback as an empty callback. 

Once we are back to alphas, we can investigate where is the best place to call it. Anyone on 1.2 can use the new callback to call it themselves manually if required.

**Comment by rlatapy-luna** [comment]
@bentrengrove Tests are not failing on my side 🤷‍♂️

**Pull request merged**